docusaurus/docs/dev-docs/migration/v4-to-v5/breaking-changes/upgrade-to-apollov4.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Apollo Server v3 upgraded to Apollo Server v4
+description: The upgrade from Apollo Server v3 to v4 and graphql ^15 to ^16.
+sidebar_label: Upgrade to Apollo Server v4
+displayed_sidebar: devDocsMigrationV5Sidebar
+tags:
+ - breaking changes
+ - Content API
+ - GraphQL
+ - GraphQL API
+---
+
+import FeedbackCallout from '/docs/snippets/backend-customization-feedback-cta.md'
+import Intro from '/docs/snippets/breaking-change-page-intro.md'
+import MigrationIntro from '/docs/snippets/breaking-change-page-migration-intro.md'
+
+# Apollo Server v3 upgraded to Apollo Server v4
+
+Strapi 5 has some changes for user code that relied on specific versions.
+<Intro />
+
+## Breaking change description
+
+<SideBySideContainer>
+
+<SideBySideColumn>
+
+**In Strapi v4**
+
+Apollo Server v3 for the GraphQL server and graphql ^15 for the GraphQL module.
+
+</SideBySideColumn>
+
+<SideBySideColumn>
+
+**In Strapi 5**
+
+Apollo Server v4 for the GraphQL server and graphql ^16 for the GraphQL module.
+
+</SideBySideColumn>
+
+</SideBySideContainer>
+
+:::callout
+Strapi automatically sets `status400ForVariableCoercionErrors: true` in the Apollo Server configuration due to a breaking change introduced by Apollo in v4. This behavior will be the default again in Apollo v5.
+:::
+
+:::callout
+We upgraded the package graphql from ^15 to ^16, and only one version can be included in any project. Therefore, user code or plugins relying on graphql:15 need to use the same version range as Strapi.
+:::
+
+## Migration
+
+<MigrationIntro />
+
+### Notes
+
+- The migration process for user code and configuration is too complicated for Strapi to cover completely. 
+- For detailed instrustion consult the Apollo v4 migration documentation.
+- To the following Manual migration section contains some key changes relevant to Strapi users.
+
+### Manual migration
+
+To migrate to Strapi 5:
```

Comment:
This section should only contain instructions and procedures. 
Most of the sentences from the bullet points that you wrote describe differences, not what the user should actually do. Rewrite the sentences that the user knows what to do. 

If you miss some information, please ask Ben Irvin, or if he's too busy at the moment, move these sentences to the Notes section and write something really generic in the Manual migration section, such as "To migrate to Strapi 5, users should manually update their code, taking into account all the items listed in the Notes section."

If you do this (i.e., do not actually describe what people can do), please also add a TODO comment in the Markdown file so we make sure we get back to it later and improve the manual migration section. Thank you!